### PR TITLE
WKO 4.0: Update ItIstioGatewaySessionMigration,ItIstioManagedCoherence,ItIstioSessionMigration, ItDedicatedMode to V9 ClusterResource

### PR DESCRIPTION
**Pull request by xiancao**
WKO 4.0: Update ItIstioGatewaySessionMigration,ItIstioManagedCoherence,ItIstioSessionMigration, ItDedicatedMode to V9 ClusterResource

Jenkins result:
https://build.weblogick8s.org:8443/view/all/job/weblogic-kubernetes-operator-kind-new/12482/

### CODE REVIEW

**Review by anpanigr — APPROVED** [review]
lgtm

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/ClusterUtils.java [hunk 1]
```diff
@@ -67,28 +67,28 @@ public static ClusterResource createClusterResource(String clusterResName,
    */
   public static void createClusterAndVerify(ClusterResource cluster) {
     LoggingFacade logger = getLogger();
-    String clusterName = cluster.getSpec().getClusterName();
+    String clusterResourceName = cluster.getClusterResourceName();
     String namespace = cluster.getNamespace();
     // create the cluster CR
     assertNotNull(cluster, "cluster is null");
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/ClusterUtils.java:73]: This is not going to fail, if cluster is null line # 70 will throw NPE
You can move line 70 and 71 below these null checks

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/ClusterUtils.java [hunk 2]
```diff
@@ -67,28 +67,28 @@ public static ClusterResource createClusterResource(String clusterResName,
...
     logger.info("Creating cluster custom resource for clusterName {0} in namespace {1}",
-        clusterName, namespace);
+        clusterResourceName, namespace);
     assertTrue(assertDoesNotThrow(() -> createClusterCustomResource(cluster),
             String.format("Create cluster custom resource failed with ApiException for %s in namespace %s",
-                clusterName, namespace)),
+                clusterResourceName, namespace)),
         String.format("Create cluster custom resource failed with ApiException for %s in namespace %s",
-            clusterName, namespace));
+            clusterResourceName, namespace));
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/ClusterUtils.java:83]: If createClusterCustomResource returns false then there is no ApiException was thrown then this assert message will be misleading.

**Review by sankarpn — CHANGES_REQUESTED** [review]
minor comments

**Review by sankarpn — APPROVED** [review]
lgtm

**Pull request merged**